src/assets/TokensController.ts

```diff
@@ -268,6 +295,68 @@ export class TokensController extends BaseController<
     }
   }
 
+  /**
+   * Adds isERC721 field to token object
+   * (Called when a user attempts to add tokens that were previously added which do not yet had isERC721 field)
+   *
+   * @param {string} tokenAddress - The contract address of the token requiring the isERC721 field added.
+   * @returns {Promise<object>} The new token object with the added isERC721 field.
+   *
+   */
+  async updateTokenType(tokenAddress: string) {
+    const { tokens } = this.state;
+    const tokenIndex = tokens.findIndex((token) => {
+      return token.address === tokenAddress;
+    });
+    tokens[tokenIndex].isERC721 = await this._detectIsERC721(tokenAddress);
```

Comment:
The token may have been deleted by the time this call resolves. Or it might be at a different index.

It would be safer to do this in a different order. Maybe by calling this first and storing the result in a local variable, then finding the token and setting the property if it's found.

Or we could use a lock, but it'd be faster to avoid locks if possible.

Reply:
addressed!